### DISCUSSION TITLE
Installing Warp

### DISCUSSION THREAD

**Discussion by elviskahoro** [warpdotdev/Warp#684] [Archived - Launched]
Discussion for where you can install Warp

**Comment by elviskahoro** [comment]
Support installing Warp from Brew

From @mittalyashu via #110

This feature can be implemented after public release of Warp.

**Reply by elviskahoro** [reply]
From @zachlloyd via #110 

Yup, we will do this once warp is GA

**Reply by elviskahoro** [reply]
Not GA yet but we support being installed from Homebrew!

**Reply by justdriveit** [reply]
First, go rob a bank, because you're gonna have to pay for it.  Then, after you have accumulated 10k for your 6 months, go to the Warp web sight, and install the program.  Be amazed at what's possible, and then realize, you could feed a small African village cheaper for a month, then use Warp on a day to day basis.  After you have determined if you want an African village, or Warp, then download.  PS. African villages can learn to code too.  And they won't constantly fail to prevent your back-up copies from being saved vs. your originals.  This seems like a sceme.  Where they set you up for failure, and keep you pining for more.  It's like gambling or drug addiction.  I ended up paying for warp, and then got caught up in the reality that it was in fact more expensive then crack.  After that, I made another account, and tried again.. To my amazement, the SAME EXACT ERROR happened right after my free service was going to end.  EXACTLY the same one... the same way, the same style... Almost like the first time, they ruined my work, and made me run back for more, and then when I tried again, they did it again... They have it set up to FK you over, to get you to run back, to finish your project.  It's amazing how that works.  Just like a Fkin slot machine.

**Reply by justdriveit** [reply]
Dun worry, I only have 200 accounts.

edit- okay, that's a lie, I have less, but I do have an internet footprint of over 10,000 pages, and a history of being super honest.  I was going to do 5 solid positive reviews for warp per account... But... I was gonna finish 1 program, and get it functional.. Did I finish?  Nope..... It was some of my own fault.. but... there were days where I did in fact spend 10 hours trying to correct 1 simple thing.. I had warp look at the files over and over and over and it kept me at 6.5GB.  And then one fine day, it magically found out it was holding a bunch of back-ups.... So it got cut down to 2.5GB... but... Ya know... There was a LOT of time spent going through that extra 4GB... Didn't do it's job... Costed me days... Costed me effort... And I'd still pay like $50 a month probably.  But as things sit, I'm one of many who are ready to kick this company in the metaphorical teeth for being so expensive.... And maybe it's just a rich man's service... But there is going to be backlash... not because everyone doesn't love it.  Because they offer it for free, then cut it off.  It's like the crack dealer who passes out rocks at the high school to get the kids addicted, then turns around and peddles his product.... And they'll probably get this account banned, and I'm cool with that.  I'm about ready to pay for Github again...... I might sell a kidney and try to get a month of warp down the line, or maybe try selling children for the cartel or something... I gotta find a way to afford this service.

**Comment by elviskahoro** [comment]
From @mittalyashu via #95

Can you make warp available on [Setapp](https://setapp.com/)?

That way we can even use warp premium features.

**Reply by elviskahoro** [reply]
From @donaldguy via #95

As a Setapp user myself, I would find appeal in this. 

But I also think, in practice, its asking Warp (or "Denver Technologies, Inc." per website footer) to unreasonably complicate their revenue model (as well as some aspects of product design and ~release management)—largely because only here in beta (while Metal is the only supported renderer) is Warp designed to be mac-exclusive app.

Warp's [stated](https://github.com/warpdotdev/warp/blob/ee65fdcf1286243d271aac6a7d58a99d6b17ee9b/README.md?plain=1#L43) plan is to have a free (gratis, libre) fairly fully-featured [cross-platform ](https://blog.warp.dev/how-warp-works/#looking-forward)client. 

They will, therefore, to survive and grow as a company (and recoup development costs), need to make up that lost-opportunity-cost from (enterprise) customers.

---

As such, planned premium features (logically, and those revealed to date) are tied to collaboration and distributed/shared workspaces. This means that each sale/subscription will have an associated COGS/opex to operate backends.

For features like infinite history (an example of one that would appeal to a Setapp-subscribing individual user), there will be a growing storage-cost over time. **_And_** that cost would be incurred every month regardless of whether or not you used Warp that month (something I have to imagine has some effect on how MacPaw splits Setapp revenue amongst apps). For good UX, it logically couldn't be vacated until you at least uninstall Warp in Setapp, and probably til you unsubscribe from Setapp entirely.

This isn't 100% untrod territory for Setapp<sup>1</sup> (c.f. [Timing's pricing](https://timingapp.com/pricing#)), and probably less so for [Setapp for Teams](https://setapp.com/teams)


But the trick, I think, for Timing and many other apps on Setapp is that they are Mac and iOS exclusive and hang all their storage and sync functions onto CloudKit et al - and so any cloud storage costs are being paid by Apple (and in turn by the end-user to Apple if they are a hefty enough iCloud user)
 

---

With the notable exceptions (to my knowledge) of Timing and [Ulysses](https://ulysses.app/get/) (maybe there are others but I'm not sure), the trend seems to be that a monthly revenue share from Setapp is traded off against offering a one time<sup>2</sup> (license file or in-app) purchase. I kinda don't see that making sense for Warp.


---

It seems like, Timing, Ulysses, Session (as an example of a freemium app that offers Setapp or an in-app subscription for pro-features), etc. are finding that they can make the pricing tradeoff make sense because of Setapp's strict per-device licensing. 

For Timing, month-to-month you'd pay $8.40 for 3 macs (and in both cases they throw in limited web app for free)
for Ulysses $7/mo ($50/yr) to use on (unlimited but in practice) a mac or two + maybe an iPad
for Session, you'd pay $6/mo ($40/yr) to use (premium features, unlimited but in practice) on a mac or two + maybe an iPhone; 

Users pay those through Apple generally, so the developers only see 70% of that.

For Setapp, developers get whatever share of $10 (for 1st + $2.50/mo additional) per mac or iOS device per month

The economics can make sense, but it seems real tight - and pretty fundamentally reliant on being Apple ecosystem exclusive, which again, Warp isn't going to be.

---

So even ignoring the Mac exclusivity issue, it seem like Warp is akin to [CleanshotX](https://cleanshot.com/pricing) - where they go in on an individual user tier with a strict (and fairly low, for images) cloud usage cap.  

But at the level above that, or with more team features, for "Cloud Pro", they've judged even Setapp for Teams' $10/user/mo too low to trade off again first party ($8/usr/mo (annual, month-to-month: $10/user/mo))

---

So, with something like infinite history, or shared env vars, Warp could use iCloud, but they'd have to make it Apple clients exclusive, or they could support Dropbox sync maybe - but they'd still be incentivized to only let you use Apple clients where Setapp can enforce licensing - or otherwise find some way to make you pay separately if you want to use Warp (premium features) from Windows/Linux/etc.

It seems unlikely its worth the overhead of maintaining a separate (closed source) fork with the Setapp runtime authorization/licensing API (see e.g. https://github.com/MacPaw/Setapp-framework though that one is iOS specific?) and non-first-party-cloud, non-enterprise-backend (e.g. iCloud) support

---
<sup>1</sup> Nor is offering otherwise free(mium) app, e.g. [Meeter](https://trymeeter.com/) and [Session](https://www.stayinsession.com/pricing). 

Open source might be unprecedented though, I thought [Numi](https://github.com/nikolaeu/numi) was a counterexample, but now I look closer at the repo it seems more ~Sublime-esque: open source plugins, but fully closed source core)

<sup>2</sup>or in practice ~roughly annual "optional" upgrade purchase, per macOS release, for e.g. Dash and Bartender
